docs/experimental/examples/oauth2/globus_app.rst

```diff
@@ -1,5 +1,203 @@
+
+            transfer_client = globus_sdk.TransferClient(app=my_app)
+            flows_client = globus_sdk.FlowsClient(app=my_app)
+
+
+Usage
+-----
+
+From this point, the app manages scope validation, token caching and routing for any
+bound clients.
+
+In the above example, listing a client's or user's flows becomes as simple as:
+
+..  code-block:: python
+
+    flows = flows_client.list_flows()["flows"]
+
+If cached tokens are missing, expired, or otherwise insufficient (e.g., the first time
+you run the script), the app will automatically initiate an auth flow to acquire new
+tokens. With a UserApp, the app will print a URL to the terminal with a prompt
+instructing a the user to follow the link and enter the code they're given back into the
+terminal. With a ClientApp, the app will retrieve tokens programmatically through a
+Globus Auth API.
+
+Once this auth flow has finished, the app will cache tokens for future use and
+invocation of your requested method will proceed as expected.
+
+
+Manually Running Login Flows
+----------------------------
+
+While your app will automatically initiate and oversee auth flows as detected, sometimes
+an author may want to explicitly control when an authorization occurs. To manually
+trigger a login flow, call ``GlobusApp.run_login_flow(...)``. This will initiate an auth
+flow requesting new tokens based on the app's currently defined scope requirements, and
+caching the resulting tokens for future use.
+
+This method accepts a single optional parameter, ``auth_params``, where a caller
+may specify additional session-based auth parameters such as requiring the use of an
+MFA token or rendering with a specific message:
+
+
+..  code-block:: python
+
+    from globus_sdk.experimental.auth_requirements_error import GlobusAuthorizationParameters
+
+    ...
+
+    my_app.run_login_flow(
+        auth_params=GlobusAuthorizationParameters(
+            session_message="Please authenticate with MFA",
+            session_required_mfa=True,
+        )
+    )
+
+
+Manually Defining Scope Requirements
+------------------------------------
+
+Globus service client classes all maintain internal list of default scope requirements
+to be attached to any bound app. These scopes represent an approximation of a
+"standard set" for each service. This list however is not sufficient for all use cases.
+
+For example, the FlowsClient defines its default scopes as ``flows:view_flows`` and
+``flows:run_status`` (read-only access). These scopes will not be sufficient for a
+script which needs to create new flows or modify existing ones. For that script, the
+author must manually attach the ``flows:manage_flows`` scope to the app.
+
+This can be done in one of two ways:
+
+#.  Through a service client initialization, using the ``app_scopes`` kwarg.
+
+    ..  code-block:: python
+
+        from globus_sdk import Scope, FlowsClient
+
+        FlowsClient(app=my_app, app_scopes=[Scope(FlowsClient.scopes.manage_flows)])
```

Comment:
I'm assuming that this can drop the `Scope` instantiation, too.

```suggestion
        from globus_sdk import FlowsClient

        FlowsClient(app=my_app, app_scopes=[FlowsClient.scopes.manage_flows])
```

Reply:
This one actually can't.

Maybe it should be able to, but for now it can't.

Reply:
> Maybe it should be able to, but for now it can't.

This is mean btw in a similar vein as the slug comment above. I'd like to discuss that behavior outside of a documentation PR. We can discuss the merits of adding that and update documentation after adding that behavior if we decide to; I don't want to block this documentation from going out on that decision though.